controllers/namespace_controller.go

```diff
@@ -277,12 +277,16 @@ func (r *NamespaceIntegrationReconciler) setupResources(ctx context.Context, req
 	// Create Default Permissions
 	for quayServiceAccountPermissionMatrixKey, quayServiceAccountPermissionMatrixValue := range QuayServiceAccountPermissionMatrix {
 
-		robotAccountResult, robotAccountErr := r.createRobotAccountAssociateToSA(ctx, request, namespace, quayClient, quayOrganizationName, quayServiceAccountPermissionMatrixKey, quayServiceAccountPermissionMatrixValue, quayName, quayHostname)
-
-		if robotAccountErr != nil {
-			return robotAccountResult, robotAccountErr
+		var g errgroup.Group
+		g.Go(func() error {
+			if _, robotAccountErr := r.createRobotAccountAssociateToSA(ctx, request, namespace, quayClient, quayOrganizationName, quayServiceAccountPermissionMatrixKey, quayServiceAccountPermissionMatrixValue, quayName, quayHostname); robotAccountErr != nil {
+				return robotAccountErr
+			}
+			return nil
+		})
+		if err := g.Wait(); err != nil {
```

Comment:
I bet you wanted to do something different here, as right now you spawn a go routine and wait until its completion, i.e. it's still a blocking call.

Reply:
right, but in this case would it not be ideal to wait for all permissions to be applied before doing a build since It helps avoid a race condition?

Reply:
this patch doesn't change how this function behaves.

```
doSomething()
```

and

```
var g errgroup.Group
g.Go(func() error {
    doSomething()
    return nil
})
g.Wait()
```

have exactly the same meaning.

 I suspect you want to run multiple goroutines at the same time and then wait for all of them.

Reply:
ah, you're right. I've moved the `Wait()` outside of for loop